modules/tensor_mechanics/doc/content/documentation/modules/tensor_mechanics/generalized_plane_strain.md

```diff
@@ -0,0 +1,78 @@
+
+The kinematical equations of the generalized plane strain problem are identical to those for the plane stress or strain problems, given as
+\begin{equation}
+\left\{\begin{matrix}
+\epsilon_{xx} \\
+\epsilon_{yy} \\
+\gamma_{xy}
+\end{matrix}\right\}
+=
+\left\{\begin{matrix}
+\frac{\partial u}{\partial x} \\
+\frac{\partial v}{\partial y} \\
+\frac{\partial v}{\partial x} + \frac{\partial u}{\partial y}
+\end{matrix}\right\}
+\end{equation}
+
+The equilibrium equations for the generalized plane strain problem in the $x$-$y$ plane are given as
+\begin{equation}
+\begin{matrix}
+\frac{\partial \sigma_{xx}}{\partial x} + \frac{\partial \sigma_{xy}}{\partial y} + \bar{f}_{x} = 0 \\
+\frac{\partial \sigma_{xy}}{\partial x} + \frac{\partial \sigma_{yy}}{\partial y} + \bar{f}_{y} = 0
+\end{matrix}
+\end{equation}
+in $A$ where $\left\{ \bar{f} \right\}$ are the body forces.
+
+A further condition is required associated with the ends of the cylindrical body, i.e. planes perpendicular to the longitudinal axis. If a constant strain is prescribed as the deformation compatibility condition in the $z$-direction
+\begin{equation}
+\epsilon_{zz} = \bar{\epsilon}_{zz}
+\end{equation}
+Alternatively, a force as the stress resultant $\bar{N}_{zz}$ in the $z$-direction can be prescribed, the condition is the equilibrium condition in $z$-direction, given as
+\begin{equation}
+N_{zz} = \int_{A}{\sigma_{zz}dA} = \bar{N}_{zz}
+\end{equation}
+The stress resultant $N_{zz}$ conjugates with constant strain $\epsilon_{zz}$.
+
+The constitutive equations in terms of stress-strain relationship are given as
+\begin{equation}
+\left\{ \begin{matrix}
+\sigma_{xx} \\
+\sigma_{yy} \\
+\sigma_{zz} \\
+\tau_{xy}
+\end{matrix} \right\}
+=
+\left[ \begin{matrix}
+C_{11} & C_{12} & C_{13} & 0 \\
+C_{12} & C_{22} & C_{23} & 0 \\
+C_{13} & C_{23} & C_{33} & 0 \\
+0 & 0 & 0 & C_{66}
+\end{matrix} \right]
+\left( \left\{ \begin{matrix}
+\epsilon_{xx} \\
+\epsilon_{yy} \\
+\epsilon_{zz} \\
+\gamma_{xy}
+\end{matrix} \right\}
+-
+\left\{ \begin{matrix}
+\alpha_{x} \\
+\alpha_{y} \\
+\alpha_{z} \\
+0
+\end{matrix} \right\} \Delta T \right)
+\end{equation}
+where $C_{ij}$ are material's stiffness coefficients using Voigt notation, $\alpha_{x}$, $\alpha_{y}$ and $\alpha_{z}$ are the material's thermal expansion coefficients and $\Delta T$ is the change in temperature.
+
+The formulation above for generalized plane strain problem shares many similarities with that of the conventional plane stress or plane strain problem. All the differences are associated with the introduction of an additional degree of freedom for the out-of-plane direction.
```

Comment:
There should be some discussion here about the MOOSE implementation and its usage. This would include a brief summary of the code objects used in GPS, with links to those objects, as well as a link to the documentation on the Action, which is where we would want to point people for usage.